### PR TITLE
fix(drag-drop): make `CDK_DROP_CONTAINER` public

### LINKED ISSUE #12213: feat(drag-drop): make `CDK_DROP_CONTAINER` token public api
#### Bug, feature request, or proposal:
feature request

#### What is the expected behavior?
That the `CDK_DROP_CONTAINER` injection token be public API.

#### What is the current behavior?
The `CDK_DROP_CONTAINER` is private api.

#### What are the steps to reproduce?
See injection token located in cdk-experimental/drag-drop here:
https://github.com/angular/material2/blob/master/src/cdk-experimental/drag-drop/drop-container.ts#L58

#### What is the use-case or motivation for changing an existing behavior?
Making the `CDK_DROP_CONTAINER` token public API is necessary in order to support creating a custom drop component which extends CdkDrop. The custom drop component will need to re-provide `CDK_DROP_CONTAINER`.

i.e
```
@Component({
  providers: [
    {provide: CDK_DROP_CONTAINER, useExisting: MyCustomListComponent},
  ],
})
export class MyCustomListComponent extends CdkDrop<ListRowItem> {}
```

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
I'm using
```
- @angular/core 6.0.7
- @angular/material 6.3.2
- @angular/cdk 6.3.2
- @angular/cdk-experimental 6.3.3
```

### PR DESCRIPTION
**Pull request by thefliik**
fixes #12213

### CODE REVIEW

File: src/cdk-experimental/drag-drop/public-api.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@
  */
 
 export * from './drop';
+export {CDK_DROP_CONTAINER} from './drop-container';
```
Comment (crisbeto) [inline src/cdk-experimental/drag-drop/public-api.ts]: Can you switch this one to `export *`?

Comment (thefliik) [reply src/cdk-experimental/drag-drop/public-api.ts]: Done

**Comment by thefliik** [comment]
@crisbeto actually, right after responding to your feedback I realized, should the `@private-api` comment be removed from the `CdkDropContainer` interface now that it's also being exported?

**Comment by crisbeto** [comment]
Sure. That tag only says whether the symbol should be displayed in the docs.

**Comment by thefliik** [comment]
👍 done

**Pull request merged**